packages/matter.js/src/net/UdpMulticastServer.ts

```diff
@@ -66,31 +66,43 @@ export class UdpMulticastServer {
         );
     }
 
-    async send(message: ByteArray, netInterface?: string) {
+    async send(message: ByteArray, netInterface?: string, uniCastTarget?: string) {
         netInterface = netInterface ?? this.netInterface;
-        const netInterfaces = netInterface !== undefined ? [netInterface] : this.network.getNetInterfaces();
-        await Promise.all(
-            netInterfaces.map(async netInterface => {
-                const { ips } = this.network.getIpMac(netInterface) ?? { ips: [] };
-                await Promise.all(
-                    ips.map(async ip => {
-                        const iPv4 = isIPv4(ip);
-                        const broadcastTarget = iPv4 ? this.broadcastAddressIpv4 : this.broadcastAddressIpv6;
-                        if (broadcastTarget == undefined) {
-                            // IPv4 but disabled, so just resolve
-                            return;
-                        }
-                        try {
-                            await (
-                                await this.broadcastChannels.get(netInterface, iPv4)
-                            ).send(broadcastTarget, this.broadcastPort, message);
-                        } catch (error) {
-                            logger.info(`${netInterface}: ${(error as Error).message}`);
-                        }
-                    }),
-                );
-            }),
-        );
+
+        // When we know the network interface and the unicast target, we can send unicast
+        if (uniCastTarget !== undefined && netInterface !== undefined) {
```

Comment:
Why do we need the net interface? If we have an IP could we let the kernel's ARP/NDP tables handle routing?

Also curious about the case where there's a unistrut with no net interface... If that's illegal maybe we should throw an exception?  As written it'd end up doing a broadcast which is probably wrong

Reply:
Depending on the user settings we open one socket per network interface the user wants us to use (ok he can limit to one, or we use "all"). With this we keep track forn an incoming message via which interface it came in and then alsoi send the response to this interface. This is especially relevant (at least from my understanding so far) for "link local ipv6" traffic where the address is only reachable via "the same" interface. nd so we use the interface to select the right outgoing socket.
it is correct that we alredy should augument the "ip" with this but I'm not fure if we do in all cases/places ...

In fact I decided to stay in this PR with the "we did that way in the past" way and do not also started this topic. We already discussed doing a refactoring here and I still agree that we should be able to get rid of the network interface info, but thats it's own and bigger topic.

Reply:
Ah right sorry didn't consider that this was part of previous conversation.

R.e. link-local, yeah, we'll need to have the "%zone-id" in the address.